File: src/main/java/leetcode/twosum_1/TwoSum.java
package leetcode.twosum_1;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class TwoSum {

    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numbers = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            numbers.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++) {
            if (numbers.containsKey(target - nums[i]) && i != numbers.get(target - nums[i])) {
                return new int[]{i, numbers.get(target - nums[i])};
            }
        }

        return new int[]{};
    }

    public int[] solution1(int[] nums, int target) {
        Map<Integer, Integer> numbers = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (numbers.containsKey(target - nums[i])) {
                return new int[]{numbers.get(target - nums[i]), i};
            }

            numbers.put(nums[i], i);
        }

        return null;
    }
}

File: src/main/java/leetcode/twosum_1/TwoSum.kt
package leetcode.twosum_1

fun twoSum(nums: IntArray, target: Int): IntArray {
    val numbers = mutableMapOf<Int, Int>()

    for ((index, num) in nums.withIndex()) {
        if (numbers.containsKey(target - num)) {
            return intArrayOf(numbers.get(target - num) ?: 0, index)
        }

        numbers.put(num, index)
    }

    return intArrayOf(0, 0)
}
